### ISSUE TITLE
Error on `entry.client` when starting a new project using `arc`

### ISSUE THREAD

**Issue by guzmonne** [remix-run/remix#198]
## Description

I'm receiving the following error on a brand new project using `Architecture (AWS Lambda)` as the deploy mechanism:

```
Warning: Expected server HTML to contain a matching <script> in <body>.
    at script
    at LiveReload
    at body
    at html
    at Document (http://localhost:3333/_static/build/root-EMKERRLV.js:20:20)
    at App
    at RemixErrorBoundary (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:1877:26)
    at RemixRoute (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:2428:3)
    at Routes (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:2412:7)
    at F (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:466:17)
    at RemixErrorBoundary (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:1877:26)
    at RemixEntry (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:2216:12)
    at RemixBrowser (http://localhost:3333/_static/build/_shared/chunk-IXE73DEJ.js:2598:42)
```

## Steps to reproduce

Start a new project using `npm init remix` and select the `Architect (AWS Lambda)` option. Then run the app using `arc sandbox` and `npm run dev` on different terminal sessions. You get the error when accessing `localhost:3333`.

**Comment by raymclee** [comment]
I have the same issue using flyio.
@guzmonne method fixed it

**Comment by raymclee** [comment]
Ok I found out the problem for me. Its is because the build is missing NODE_ENV=production, so the LiveReload component still rendering on flyio

**Comment by GiancarlosIO** [comment]
Same here :(

**Comment by cbschuld** [comment]
@guzmonne 's solution also worked for me; also unsure of the implications.  Also had to change the websocket port but works as intended in the `useEffect` wrapper.

**Comment by nickbreaton** [comment]
I was able to avoid workaround this by adding a `preferences.arc` file with the following contents:

```
@env
testing
  NODE_ENV development
```

I think Remix needs to find a way to propagate the current `NODE_ENV` value into Architects sandbox server. [Only so many environment variables are carried across to the emulated Lambda.](https://github.com/architect/sandbox/blob/6c79a0f8ca06ac44d910e64c8307b68ffbe6505c/src/invoke-lambda/env/index.js#L36)

**Comment by lpsinger** [comment]
This is fixed by #6757.
